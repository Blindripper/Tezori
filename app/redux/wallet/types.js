export const SET_WALLET = 'SET_WALLET';
export const SET_WALLET_FILENAME = 'SET_WALLET_FILENAME';
export const SET_WALLET_LOCATION = 'SET_WALLET_LOCATION';
export const SET_PASSWORD = 'SET_PASSWORD';
export const SET_IDENTITIES = 'SET_IDENTITIES';
export const LOGOUT = 'LOGOUT';