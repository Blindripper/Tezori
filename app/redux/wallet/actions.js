import {
  SET_WALLET,
  SET_IS_LOADING,
  SET_WALLET_FILENAME,
  SET_WALLET_LOCATION,
  SET_PASSWORD,
  SET_IDENTITIES,
  ADD_NEW_IDENTITY,
  UPDATE_IDENTITY,
  ADD_NEW_ACCOUNT
  LOGOUT,
} from './types';
import actionCreator from '../../utils/reduxHelpers';


export const setWallet = actionCreator(SET_WALLET, 'wallet');
export const setIsLoading = actionCreator(SET_IS_LOADING, 'isLoading');
export const setWalletFileName = actionCreator( SET_WALLET_FILENAME, 'walletFileName' );
export const updateWalletLocation = actionCreator( SET_WALLET_LOCATION, 'walletLocation' );
export const setPassword = actionCreator(SET_PASSWORD, 'password');
export const setIdentities = actionCreator(SET_IDENTITIES, 'identities');
export const addNewIdentity = actionCreator(ADD_NEW_IDENTITY, 'identity');
export const updateIdentity = actionCreator(UPDATE_IDENTITY, 'identity');
export const addNewAccount = actionCreator( ADD_NEW_ACCOUNT, 'publicKeyHash', 'account' );
export const logout = actionCreator(LOGOUT);
