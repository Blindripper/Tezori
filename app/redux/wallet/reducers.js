import { fromJS } from 'immutable';
import {
  SET_WALLET,
  SET_WALLET_FILENAME,
  SET_WALLET_LOCATION,
  SET_PASSWORD,
  SET_IDENTITIES,
  LOGOUT,
} from './types';

const initState = fromJS({
  identities: [],
  walletFileName: '',
  walletLocation: '',
  password: ''
});

export default function wallet(state = initState, action) {
  switch (action.type) {
    case SET_WALLET:
      return fromJS(action.wallet);
    case SET_WALLET_FILENAME:
      return state.set('walletFileName', action.walletFileName);
    case SET_WALLET_LOCATION:
      return state.set('walletLocation', action.walletLocation);
    case SET_PASSWORD:
      return state.set('password', action.password);
    case SET_IDENTITIES:
      return state.set('identities', fromJS(action.identities));
    case LOGOUT:
      return initState;
    default:
      return state;
  }
}