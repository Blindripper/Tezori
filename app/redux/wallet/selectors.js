import { createSelector } from 'reselect';
export const getWallet = state => state.wallet;

export const getWalletName = createSelector(
  getWallet,
  wallet => {
    const fileName = wallet.get('walletFileName');
    const walletName = fileName.split('.');
    return walletName[0];
  }
);

export const getWalletIsLoading = createSelector(
  getWallet,
  wallet => wallet.get('isLoading')
);