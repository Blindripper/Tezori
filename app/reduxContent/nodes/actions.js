import {
  SET_SELECTED,
  ADD_NODE,
  REMOVE_NODE,
  UPDATE_NODE,
  CLEAR_STATE
} from './types';
import actionCreator from '../../utils/reduxHelpers';

export const setSelected = actionCreator(SET_SELECTED, 'selected', 'target');
export const addNode = actionCreator(ADD_NODE, 'node');
export const removeNode = actionCreator(REMOVE_NODE, 'name');
export const updateNode = actionCreator(UPDATE_NODE, 'node');
export const clearState = actionCreator(CLEAR_STATE);
