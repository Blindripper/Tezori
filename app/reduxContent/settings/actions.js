import {
  SET_SELECTED,
  SET_LOCAL,
  ADD_NODE,
  REMOVE_NODE,
  UPDATE_NODE,
  CLEAR_STATE
} from './types';
import actionCreator from '../../utils/reduxHelpers';

export const setSelected = actionCreator(SET_SELECTED, 'selected', 'target');
export const setLocal = actionCreator(SET_LOCAL, 'local');
export const addNode = actionCreator(ADD_NODE, 'node');
export const removeNode = actionCreator(REMOVE_NODE, 'name');
export const updateNode = actionCreator(UPDATE_NODE, 'node');
export const clearState = actionCreator(CLEAR_STATE);
