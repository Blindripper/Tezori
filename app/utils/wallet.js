import path from 'path';
import { TezosWallet } from 'conseiljs';
const { saveWallet } = TezosWallet;
export async function saveUpdatedWallet(identities, walletLocation, walletFileName, password) {
  const completeWalletPath = path.join(walletLocation, walletFileName);

  const identities = identities
    .map(({ publicKey, privateKey, publicKeyHash }) => {
      return { publicKey, privateKey, publicKeyHash };
    });

  return await saveWallet( completeWalletPath, { identities }, password );
}