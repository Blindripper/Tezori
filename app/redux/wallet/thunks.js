import path from 'path';
import { TezosWallet } from 'conseiljs';
import { push } from 'react-router-redux';
import { clearEntireAddressState } from '../../reducers/address.duck';
import { addMessage } from '../../reducers/message.duck';
import { displayError } from '../../utils/formValidation';
import { CREATE, IMPORT } from '../../constants/CreationTypes';

import {
  clearWalletState,
  setWallet
} from './actions';

const { createWallet, loadWallet, saveWallet } = TezosWallet;

export function goHomeAndClearState() {
  return dispatch => {
    dispatch(clearWalletState());
    dispatch(clearEntireAddressState());
    dispatch(push('/'));
  };
}

export function saveUpdatedWallet(identities) {
  return async (dispatch, state) => {
    try {
      const identities = state().address.get('identities').toJS();
      const walletLocation = state().wallet.get('walletLocation');
      const walletFileName = state().wallet.get('walletFileName');
      const walletIdentities = state().wallet.get('identities').toJS();

      const indices = walletIdentities.map( identity => identity.publicKeyHash );
      const password = state().wallet.get('password');
      const completeWalletPath = path.join(walletLocation, walletFileName);


      /* making sure only unique identities are added */
      const newIdentities = identities
        .filter(({ publicKeyHash }) =>
          indices.indexOf(publicKeyHash) === -1
        )
        .map(({ publicKey, privateKey, publicKeyHash }) => {
          return { publicKey, privateKey, publicKeyHash };
        });


      await saveWallet(
        completeWalletPath,
        {
          identities: walletIdentities.concat(newIdentities)
        },
        password
      );
    } catch (e) {
      dispatch(addMessage(e.name, true));
      throw e;
    }
  };
}

export function login(loginType, walletLocation, walletFileName, password, confirmedPassword) {
  return async (dispatch, state) => {
    const completeWalletPath = path.join(walletLocation, walletFileName);

    // TODO: clear out message bar
    dispatch(addMessage('', true));

    let validations = [
      { value: walletLocation, type: 'locationFilled'},
      { value: `${walletLocation}/${walletFileName}`, type: 'validJS'},
      { value: password, type: 'notEmpty', name: 'Password' },
      { value: password, type: 'minLength8', name: 'Password' }
    ];

    if (loginType === CREATE ) {
      validations.push({
        value: [password, confirmedPassword],
        type: 'samePassPhrase',
        name: 'Passwords'
      })
    }

    const error = displayError(validations);
    if (error) {
      return dispatch(addMessage(error, true));
    }

    try {
      let identities = [];
      if (loginType === CREATE) {
        identities = await createWallet(completeWalletPath, password);
      } else if (loginType === IMPORT) {
        identities = await loadWallet(completeWalletPath, password).catch((err) => {
          err.name = 'Invalid wallet/password combination.';
          throw err;
        });
      }
      
      dispatch(
        setWallet({
          identities,
          walletLocation,
          walletFileName,
          password
        }, 'wallet')
      );
    } catch (e) {
      dispatch(addMessage(e.name, true));
      throw e;
    }
  };
}