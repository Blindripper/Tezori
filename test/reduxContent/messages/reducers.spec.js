import messages from '../../../app/reduxContent/message/reducers';
import * as actions from '../../../app/reduxContent/message/actions';
import { logout } from '../../../app/reduxContent/wallet/actions';


describe('action type CLEAR_MESSAGE_STATE', () => {
    test('returns the correct state', () => {  
      expect(messages(undefined, actions.clearMessageState())).toMatchSnapshot();
    });
  });

  describe('action type ADD_MESSAGE', () => {
    test('returns the correct state', () => {  
      expect(messages(undefined, actions.createMessage('test',true))).toMatchSnapshot();
    });
  });

  describe('action type LOGOUT', () => {
    test('returns the correct state', () => {  
      expect(messages(undefined, logout())).toMatchSnapshot();
    });
  });