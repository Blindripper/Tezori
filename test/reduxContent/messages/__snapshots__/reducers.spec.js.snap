// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action type ADD_MESSAGE returns the correct state 1`] = `
Immutable.Map {
  "message": Immutable.Map {
    "message": "test",
    "isError": true,
  },
}
`;

exports[`action type CLEAR_MESSAGE_STATE returns the correct state 1`] = `
Immutable.Map {
  "message": Immutable.Map {},
}
`;

exports[`action type LOGOUT returns the correct state 1`] = `
Immutable.Map {
  "message": Immutable.Map {},
}
`;
